libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -34,21 +34,22 @@ void ImmutableValidator::analyze()
 	for (ContractDefinition const* contract: linearizedContracts)
 		for (VariableDeclaration const* stateVar: contract->stateVariables())
 			if (stateVar->value())
-				m_initializedStateVariables.emplace(stateVar);
+				stateVar->value()->accept(*this);
 
 	for (ContractDefinition const* contract: linearizedContracts)
-		for (VariableDeclaration const* stateVar: contract->stateVariables())
-			if (stateVar->value())
-				stateVar->value()->accept(*this);
+		for (std::shared_ptr<InheritanceSpecifier> const& inheritSpec: contract->baseContracts())
+			if (auto args = inheritSpec->arguments())
+				ASTNode::listAccept(*args, *this);
 
 	for (ContractDefinition const* contract: linearizedContracts)
+	{
 		if (contract->constructor())
 			visitCallableIfNew(*contract->constructor());
 
-	for (ContractDefinition const* contract: linearizedContracts)
-		for (std::shared_ptr<InheritanceSpecifier> const& inheritSpec: contract->baseContracts())
-			if (auto args = inheritSpec->arguments())
-				ASTNode::listAccept(*args, *this);
+		for (VariableDeclaration const* stateVar: contract->stateVariables())
```

Comment:
I think in both versions, the state variables are initialized before the constructor is run.

Reply:
Please cross-check with https://github.com/ethereum/solidity/blob/develop/docs/ir/ir-breaking-changes.rst